File: templates/go/product.go
package main

import (
  "fmt"
  "net/http"
  "io"
)

func main() {
  resp, err := http.Get("http://localhost:3001/api/product")
  if err != nil {
    fmt.Println("Error:", err)
    return
  }
  defer resp.Body.Close()
  body, err := io.ReadAll(resp.Body)
  if err != nil {
    fmt.Println("Error:", err)
    return
  }
  fmt.Println("Product:", string(body))
}

File: templates/rust/product.rs
use reqwest;

#[tokio::main]
async fn main() -> Result<(), reqwest::Error> {
    let res = reqwest::get("http://localhost:3001/api/product").await?;
    let text = res.text().await?;
    println!("Product: {}", text);
    Ok(())
}